crypto/multisig/src/lib.rs

```diff
@@ -146,35 +187,47 @@ impl<P: Default + PublicKey + Message> SignerSet<P> {
         // While the verification code below should be immune to duplicate signers or
         // signatures, the overhead of deduping them is negligible and being
         // extra-safe is a good idea.
-        let mut potential_signers = self.signers.clone();
-        potential_signers.sort();
-        potential_signers.dedup();
+        let mut potential_individual_signers = self.individual_signers.clone();
+        potential_individual_signers.sort();
+        potential_individual_signers.dedup();
+
+        let mut potential_multi_signers = self.multi_signers.clone();
+        potential_multi_signers.sort();
+        potential_multi_signers.dedup();
 
         let mut signatures = multi_sig.signatures.clone();
         signatures.sort_by(|a, b| a.as_ref().cmp(b.as_ref()));
         signatures.dedup();
```

Comment:
:wrench: This is cloning, sorting and deduping for each subsequent set.
Probably not much of a performance concern with the max signature count of 10.

Reply:
Correct. The solution to overcome this is to add a private helper method that accepts an already-sorted signatures list. I opted against because I think it's a bit more readable this way, but that is easy to change.